### PR TITLE
Dependency, bundling, and PR deploy fixes

### PR DESCRIPTION
**Pull request by ecraig12345**
Dependency fixes:
- Fix some security alerts by updating dependencies.
- Remove `react-text` dependency on `@fluentui/storybook` which was probably causing it to build more packages than necessary.
- Remove unneeded dependencies on `example-app-base` and `prop-types`.

Removing the deps on `example-app-base` caused bundling the demo apps for PR deploy to stop working. The issue turned out to just be that we needed to be calculating resolve aliases for *legacy* demo apps relative to `react-examples`, like we were already doing with storybook demos, but there were a lot of red herrings and rabbit holes on the path to realizing that. Resulting in this overall good (but more time consuming) list of fixes:
- Add more helpers to de-duplicate the webpack configs of component packages: `createBundleConfig` and `createLegacyDemoAppConfig`. (This made it easy to do the fix for the actual issue in one place.)
- Move build output of the demo apps (both legacy and storybook) into sub-directories of `dist`: `dist/storybook` and `dist/demo`
- Stop publishing demo apps and react in our npm packages!! (This was happening in any package with a legacy demo app.) Putting the demo apps under `dist/demo` made this much easier.

Also PR deploy site improvements:
- Sort the packages in the PR deploy site
- Make the message in the header work for branch CI builds, not just PR builds
- Add instructions for how to add new things to the PR deploy site

Other cleanup/improvements:
- Delete `react-spinbutton` for now since we probably won't be using it in the immediate future (can re-create with current template when needed).
- Delete webpack and copy configs which aren't needed
- Remove react-cards legacy demo app which isn't really needed
- Delete empty or incorrectly copy-pasted doc sections for secondary packages (charting, experiments)
- Remove old version of apps/perf-test which isn't being used (the flamegrill-based part is still there)
- Remove the todo app from the PR deploy site since it's not very useful to have there
- Update azure pipelines to use more shared variables

### CODE REVIEW

**Review by xugao — DISMISSED** [review]

**Comment by ecraig12345** [comment]
@xugao Please take another look at this when you get a chance, since it changed significantly (see the updated PR description).

File: scripts/webpack/webpack-resources.js [hunk 1]
```diff
@@ -221,33 +297,56 @@ module.exports = {
...
+      throw new Error(`${packageName} does not have a legacy demo app (expected location: ${demoEntryInExamples})`);
+    }
+
+    return module.exports.createServeConfig({
+      entry: {
+        'demo-app': demoEntryInExamples,
+      },
+
+      output: {
+        path: path.join(process.cwd(), 'dist/demo'),
```
Comment (xugao) [inline scripts/webpack/webpack-resources.js:334]: this is not quite right; http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/15414/merge/public-docsite-resources/demo/demo-app.js is 404

File: apps/pr-deploy-site/pr-deploy-site.js [hunk 2]
```diff
@@ -3,88 +3,118 @@
...
-    link: './public-docsite-resources/index.html',
+    link: './public-docsite-resources/demo/index.html',
     icon: 'FavoriteStar',
-    title: 'Demo Site',
+    title: '@fluentui/react demo',
   },
   {
     package: '@fluentui/react',
-    link: './react/dist-storybook/index.html',
+    link: './react/storybook/index.html',
```
Comment (xugao) [inline apps/pr-deploy-site/pr-deploy-site.js:12]: all the stories are not loading for some reason

File: apps/pr-deploy-site/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+These steps cover the most common scenario, a component package with a storybook-based demo site with its examples in `react-examples`. In these steps, we'll call the new package `@fluentui/react-notbutton` (substitute the real name).
+
+1. Ensure the package has examples under `packages/react-examples/src/react-notbutton`
+2. In `apps/pr-deploy-site/just.config.ts`, add `@fluentui/react-notbutton` to the `dependencies` array.
+3. In `apps/pr-deploy-site/pr-deploy-site.js`, add an entry for `@fluentui/react-notbutton`, substituting actual appropriate values. (Choose an icon name [from this page](https://developer.microsoft.com/en-us/fluentui#/styles/web/icons#available-icons).)
+   ```ts
+   {
+     package: '@fluentui/react-notbutton',
+     link: './react-notbutton/dist/storybook/index.html',
```
[the remark this answers is not in the document]
Comment (xugao) [reply apps/pr-deploy-site/README.md:17]: also, LOL on `notbutton` :p

[the remark this answers is not in the document]
Comment (ecraig12345) [reply apps/pr-deploy-site/README.md:17]: I mean...we had to get it in there somehow 🤣

File: scripts/webpack/webpack-resources.js (hunk 1 above)
Comment (ecraig12345) [reply scripts/webpack/webpack-resources.js:334]: I briefly got excited upon seeing the green builds...but yup the site is broken. I just deleted the existing version from blob storage so I can run a new PR build and fully see what it's doing.

File: azure-pipelines.yml [hunk 4]
```diff
@@ -92,11 +92,12 @@ jobs:
       - task: AzureUpload@2
         displayName: Upload PR deploy site
         inputs:
+          azureSubscription: $(azureSubscription)
+          BlobPrefix: $(deployBasePath)
+          CacheControl: 'public, max-age=600000'
```
Comment (xugao) [inline azure-pipelines.yml:100]: hm shouldn't this be no-cache?

Comment (ecraig12345) [reply azure-pipelines.yml:100]: I think I'll just remove it again.

**Comment by ecraig12345** [comment]
Everything should be working now, except perf-test deploy (which is also broken in master) and todo app (which I removed from the site because I feel like it's not that useful to have). perf-test is just showing a bunch of avatars for some reason, but it's doing that in master as well, so that can be fixed separately.

**Review by xugao — APPROVED** [review]
Thanks for making all these improvements! 
Regarding perf-test, everything looks right to me. So I'm not sure what you meant by not working (we can follow up offline).

**Comment by ecraig12345** [comment]
Oh I just meant that the link on the deployed site doesn't seem to be showing anything very useful? http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/15414/merge/perf-test/index.html

**Pull request merged**